### PR TITLE
Add Release Workflows

### PR DESCRIPTION
**Pull request by bbazukun123**
- Add `on-release` workflows:
     - `release.yml`: package and add the packaged code to the version release.
     - `publish.yml`: publish the package to NPM

### CODE REVIEW

**Review by achamas-playco — APPROVED** [review]

**Review by bigtimebuddy — CHANGES_REQUESTED** [review]
Overall, I think this is little bit complicated for the scope of this package.

I don't think we need to publish an tgz artifact to the release. In general, we use release artifacts for things like browser bundles, but if someone needs a tarball for any reason, download it from the npm registry. It doesn't add a ton of value and is not really consistent with how we do any of our other packages.

The other missed opportunity with this workflow is to run build/lint/etc on PRs and branch pushes. Would be a great way to validate PRs. These workflows are only doing this on releases, which mean that a blocking build or lint issue could happen too late and your release is blocked.

My suggestion is to create *one* workflow that does a build and optionally publishes for a release. For example:

```yml
name: Build
on:
    push:
        branches: ['**']
    release:
        types: [published]
    pull_request:
        branches: ['**']
jobs:
    build:
        runs-on: ubuntu-latest
        steps:
            - uses: actions/checkout@v3
            - name: Use Node.js 18.x
              uses: actions/setup-node@v3
              with:
                  node-version: 18
                  cache: 'npm'
            - name: Install Dependencies
              run: npm ci
            - name: Build
              run: npm run build
            - name: Publish Release
              if: github.event_name == 'release'
              run: npm publish
              env:
                  NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```

So in order to publish, you do `npm version` locally (i usually add a postversion script to `git push && git push --tags`), and then create a GitHub release from that tag.

**Comment by bbazukun123** [comment]
> Overall, I think this is little bit complicated for the scope of this package.
> 
> I don't think we need to publish an tgz artifact to the release. In general, we use release artifacts for things like browser bundles, but if someone needs a tarball for any reason, download it from the npm registry. It doesn't add a ton of value and is not really consistent with how we do any of our other packages.
> 
> The other missed opportunity with this workflow is to run build/lint/etc on PRs and branch pushes. Would be a great way to validate PRs. These workflows are only doing this on releases, which mean that a blocking build or lint issue could happen too late and your release is blocked.
> 
> My suggestion is to create _one_ workflow that does a build and optionally publishes for a release. For example:
> 
> ```yaml
> name: Build
> on:
>     push:
>         branches: ['**']
>     release:
>         types: [published]
>     pull_request:
>         branches: ['**']
> jobs:
>     build:
>         runs-on: ubuntu-latest
>         steps:
>             - uses: actions/checkout@v3
>             - name: Use Node.js 18.x
>               uses: actions/setup-node@v3
>               with:
>                   node-version: 18
>                   cache: 'npm'
>             - name: Install Dependencies
>               run: npm ci
>             - name: Build
>               run: npm run build
>             - name: Publish Release
>               if: github.event_name == 'release'
>               run: npm publish
>               env:
>                   NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
> ```
> 
> So in order to publish, you do `npm version` locally (i usually add a postversion script to `git push && git push --tags`), and then create a GitHub release from that tag.

The `release` workflow has been removed and the `publish` workflow has been replaced with the suggestion above. With the lint updates, these should come nicely together 🤞

File: package.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
     "lint": "eslint src --ext .ts --max-warnings 0",
     "types": "tsc",
     "postversion": "git push && git push --tags",
-    "prepack": "npm run build && clean-package",
+    "prepare": "npm run build",
```
Comment (bigtimebuddy) [inline package.json:14]: This is unnecessary as the workflow already handles building.

```suggestion
```

**Review by bigtimebuddy — APPROVED** [review]
Minor thing to prevent building twice on CI

File: package.json (hunk 1 above)
Comment (bbazukun123) [reply package.json:14]: woops, removed 👍 Thx

**Pull request merged**